internal/cow/cow.go

```diff
@@ -90,4 +90,7 @@ type Container interface {
 	Wait() error
 	// Modify sends a request to modify container resources
 	Modify(ctx context.Context, config interface{}) error
+
+	// WaitChannel returns the wait channel of the container
+	WaitChannel() <-chan struct{}
```

Comment:
Can we add a `WaitErr() err` that returns the same error that `Wait` would have returned?

Reply:
and then it would be nice if the `Wait() error` implementation can just become:

```golang
<-c.WaitChannel()
return c.WaitError()
```

Reply:
Do we not plan to refactor/clean up ctr lifecycles? 
In internal/gcs/container.go, Wait() is as follows:

func (c *Container) Wait() error {
	select {
	case <-c.notifyCh:
		return nil
	case <-c.closeCh:
		return errors.New("container closed")
	}
}

I do not see closeCh being used anywhere (doesn't get closed either). I could cleanup as part of this fix and refactor Wait() in this file like you suggested. Or should we come back to this and refactor our ctr lifecycle code later? Thoughts?

Reply:
I don't think we should do broader refactoring right now, and I think rather than removing `closeCh` we should actually close it in `Close`, which is probably what was originally intended. But I think adding `WaitChannel`/`WaitError` functions as a non-blocking alternative to `Wait` is a good plan.